cmd/utils/flags.go

```diff
@@ -424,12 +426,25 @@ func MakeEthConfig(clientID, version string, ctx *cli.Context) *eth.Config {
 	if err != nil {
 		glog.V(logger.Error).Infoln("WARNING: No etherbase set and no accounts found as default")
 	}
+	// Resolve the mode of opeation from the string flag
+	var clientMode eth.Mode
+	switch strings.ToLower(ctx.GlobalString(EthModeFlag.Name)) {
+	case "archive":
+		clientMode = eth.ArchiveMode
+	case "full":
+		clientMode = eth.FullMode
+	case "light":
+		clientMode = eth.LightMode
+	default:
+		glog.Fatalf("Unknown node type requested: %s", ctx.GlobalString(EthModeFlag.Name))
+	}
```

Comment:
The whole backend is in eth, I definitely need that, I think duplicating the backend would be wrong. I just won't construct an eth ProtocolManager (I do agree about dropping NoSync, it was a bad idea). I can drop the mode flag and add a bool flag for light mode of course.